docs/pages/access-controls/access-requests/oss-role-requests.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: Just-in-time Access Requests in OSS Teleport
+description: OSS Teleport allows users to request access to roles from the CLI.
+h1: Teleport Resource Access Requests
+---
+
+Just-in-time Access Requests are a feature of Teleport Enterprise.
+
+Open-source Teleport users can get a preview of how Access Requests work by
+requesting a role via the Teleport CLI. Full Access Request functionality,
+including Resource Access Requests and an intuitive and searchable UI are
+available in Teleport Enterprise.
+
+## RBAC Setup
+
+Teleport's role-based access control (RBAC) allows you to configure what roles
+users can request access to. In this example, we will define two roles:
+
+1. `contractor`: users with this role can request elevated access to the `dba` role
+2. `dba`: this role grants access to databases
+
+There is no role for request approvers, because request approval rules can only
+be configured for Teleport Enterprise. In Open Source Teleport, approvals must
+be performed by running `tctl` on the Auth Server.
+
+**Contractor Role**
+
+Users with this role can request access to the `dba` role.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: contractor
+spec:
+  allow:
+    request:
+      roles: ['dba']
+```
+
+**DBA Role**
+
+This role grants access to databases.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: dba
+spec:
+  allow:
+    db_labels:
+      '*': '*'
+  options:
+    # Only allows the contractor to use this role for 1 hour from time of request.
+    max_session_ttl: 1h
+```
+
+## Requesting Access
```

Comment:
I believe we're missing a step here to add the contractor role to a user before it requests access to the dba role. You can add something like this:

```
tctl users update --set-roles $(tctl get users/alice --format=json | \
jq -r '.[].spec.roles | join(",")'),contractor alice
```

Reply:
Nifty jq snippet, maybe I'll get around to implementing `tctl users update --add-roles` at some point so we can simplify it.

Reply:
At the very least, how about a warning that running `users update` will remove any existing attributes not specified in the command?